Add docs for fusion strategy

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #72077
* __->__ #72036
* #72038
* #72040
* #72032
* #71651
* #71650
* #71173

Differential Revision: [D33864651](https://our.internmc.facebook.com/intern/diff/D33864651)

Commit:
Add docs for fusion strategy

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33864651).

Commit:
Update on "Add docs for fusion strategy"


Differential Revision: [D33864651](https://our.internmc.facebook.com/intern/diff/D33864651)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33864651).

Commit:
Update on "Add docs for fusion strategy"


Differential Revision: [D33864651](https://our.internmc.facebook.com/intern/diff/D33864651)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33864651).

navahgar commented on torch/csrc/jit/runtime/profiling_graph_executor_impl.h:
```diff
@@ -7,9 +7,51 @@ namespace jit {
+// Behavior - static vs dynamic:
+// - in STATIC fusion, fused ops are compiled to have fixed input shapes. The
+//   input shapes are determined based on a number of initial profiling runs.
+//   The shape is determined based on some initial profiling runs. For example,
```
Please clarify what shape is referred to here.

eellison replied:
i think this is used throughout pytorch code base, e.g. 
![image](https://user-images.githubusercontent.com/11477974/151864827-97fd76d6-d31f-4482-9f7b-006faba76505.png)

navahgar reviewed (approved):
LGTM

Commit:
Update on "Add docs for fusion strategy"


Differential Revision: [D33864651](https://our.internmc.facebook.com/intern/diff/D33864651)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33864651).
